library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerDownloader.java

```diff
@@ -55,6 +57,7 @@
     public void download(Batch batch, Map<DownloadBatchId, DownloadBatch> downloadBatchMap) {
         DownloadBatch runningDownloadBatch = downloadBatchMap.get(batch.downloadBatchId());
         if (runningDownloadBatch != null) {
+            Log.w("Ignoring duplicate batch with id " + batch.downloadBatchId());
```

Comment:
This might be triggered by `submitAllStoredDownloads` right? Would it be better to say:

"Already running download for DownloadBatchId: %s , ensure you are not duplicating identifiers."

Reply:
No, that method can only be called by the client using the interface

Reply:
What you are referring to is the next one, with different signature (taking a `DownloadBatch`).
But we'll never reach that point: batches are filtered first